### DISCUSSION TITLE
Play connection pool with pgbouncer

### DISCUSSION THREAD

**Discussion by Knitter** [playframework/playframework#12350] [Q&A] [answered]
Hi,

I currently have a PostgreSQL database, with pgBouncer providing connection pool to all clients (desktop, web, mobile, etc.). Now I'll be adding some Play based applications; since play already has a connection pool, it is my understanding that I shouldn't connect Play to pgBouncer. Is this correct?

Having two pools connected to each other will, I assume, cause issues when Play holds onto a connection and pgBouncer wants to release it to some other client, or am I missing this completely?

Thanks.

**Comment by irumiha** [comment, marked as answer]
According to this: https://github.com/brettwooldridge/HikariCP/issues/1042#issuecomment-1034620052 you can make it work. Also check out the rest of that thread for some additional experience people had with such setups.

**Comment by Knitter** [comment]
Hi,

I appreciate your answers, I'll be doing some tests to see how using both pools affects our systems, but most likely I'll setup Play to connect to the database server without pgBouncer, and keep pgBouncer for other clients like desktop applications.

Thanks.
